### PR TITLE
Dev 037 starring products

### PR DESCRIPTION
**Pull request by Paul-Gerarts**
Don't forget to test the new functionalities!

### CODE REVIEW

File: src/main/java/be/syntra/devshop/DevshopFront/controllers/ProductController.java [hunk 1]
```diff
@@ -50,22 +56,29 @@ public String displayProductOverview(Model model) {
     }
 
     @GetMapping("/details/{id}")
-    public String handleGet(@PathVariable Long id, Model model) {
-        Product product = productService.findById(id);
-        model.addAttribute("product", product);
-        model.addAttribute("cart", cartService.getCart());
-        return "product/productDetails";
+    public String handleGet(@PathVariable Long id, Model model, Principal user) {
```
Comment (lenshuygh) [inline src/main/java/be/syntra/devshop/DevshopFront/controllers/ProductController.java:59]: this method name bothers me every time i see it, i would suggest renaming it to 'getProductDetails'.... but that one is already taken...

File: src/main/java/be/syntra/devshop/DevshopFront/models/StatusNotification.java [hunk 2]
```diff
@@ -18,7 +18,9 @@
     DELETE_FAIL("deletion failed"),
     PAYMENT_FAIL("Payment cart failed."),
     UPDATE_FAIL("Failed to change the category"),
-    CATEGORY_EXISTS("Creation failed. Category already excists");
+    RATING_FAIL("Failed to submit the rating"),
+    CATEGORY_EXISTS("Creation failed. Category already excists"),
```
Comment (lenshuygh) [inline src/main/java/be/syntra/devshop/DevshopFront/models/StatusNotification.java:22]: 'exists'

File: src/main/java/be/syntra/devshop/DevshopFront/services/StarRatingServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @PostConstruct
+    private void init() {
+        resourceUrl = baseUrl.concat(endpoint);
+    }
+
+    @Override
+    public StarRatingDto findBy(Long productId, String userName) {
```
Comment (lenshuygh) [inline src/main/java/be/syntra/devshop/DevshopFront/services/StarRatingServiceImpl.java:39]: 'findByUserAndId' ?

File: src/main/java/be/syntra/devshop/DevshopFront/controllers/ProductController.java (hunk 1 above)
Comment (Paul-Gerarts) [reply src/main/java/be/syntra/devshop/DevshopFront/controllers/ProductController.java:59]: life is simple :) -> getProductDetailsById

File: src/main/java/be/syntra/devshop/DevshopFront/models/StatusNotification.java (hunk 2 above)
Comment (Paul-Gerarts) [reply src/main/java/be/syntra/devshop/DevshopFront/models/StatusNotification.java:22]: fixed

File: src/main/java/be/syntra/devshop/DevshopFront/services/StarRatingServiceImpl.java (hunk 3 above)
Comment (Paul-Gerarts) [reply src/main/java/be/syntra/devshop/DevshopFront/services/StarRatingServiceImpl.java:39]: findByUserNameAndId

**Pull request merged**